audio/audiodata.c

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2017 Kyle Robbertze <krobbertze@gmail.com>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+/** \file audiodata.c
+ * Contains definitions for USB audio communication between the FX2 chip and
+ * the host
+ */
+
+#include <stdio.h>
+#include <fx2regs.h>
+#include <delay.h>
+#include <eputils.h>
+
+#include "audiodata.h"
+
+#define SYNCDELAY SYNCDELAY4
+
+/**
+ * Returns the configuration. We only support cfg 1
+ */
+BYTE handle_get_configuration() {
+    return 1;
+}
+
+/**
+ * Sets the configuration. Successful if setting it to cfg 1, otherwise fail
+ * as that is the only one supported
+ */
+BOOL handle_set_configuration(BYTE cfg) {
+    return cfg==1 ? TRUE : FALSE;
+}
+
+/* The current alternative setting */
+BYTE alt_setting = 0;
+/**
+ * Returns the interface currently in use.
+ */
+BOOL handle_get_interface(BYTE ifc, BYTE* alt_ifc) {
+    printf("Get Interface\n");
```

Comment:
Can you change the printf's to work like the ones here -> https://github.com/mithro/fx2lib/blob/master/lib/setupdat.c#L21-L26
```c
#ifdef DEBUG_SETUPDAT
#include <stdio.h> // NOTE this needs deleted
#else
#define printf(...)
#define NULL (void*)0;
#endif
```

Reply:
Fixed